app/utils/injectSaga.js

```diff
@@ -27,21 +27,33 @@ export default ({ key, saga, mode }) => WrappedComponent => {
       WrappedComponent.name ||
       'Component'})`;
 
-    componentWillMount() {
-      const { injectSaga } = this.injectors;
+    state = {
+      isReady: false,
+    };
 
-      injectSaga(key, { saga, mode }, this.props);
+    componentDidMount() {
+      const { injectSaga } = this.injectors;
+      this.isComponentMounted = true;
+      injectSaga(key, { saga, mode }, this.props).then(() => {
+        if (this.isComponentMounted) {
+          this.setState({ isReady: true });
+        }
```

Comment:
I don't understand. What's the purpose of `this.isComponentMounted`? In case the component is unmounted before the injectSaga promise is resolved?

In any case, lines 38-40 would look cleaner if you did:

```js
this.setState({ isReady: this.isComponentMounted });
```